handlers/user.go

```diff
@@ -512,3 +513,71 @@ func (h *handler) ListFavouriteSongs(w http.ResponseWriter, r *http.Request) {
 
 	jsonResponse(w, topTracks.Tracks)
 }
+
+type SyncMode string
+
+const (
+	ForceSync   SyncMode = "FORCE_SYNC"
+	ForceDesync SyncMode = "FORCE_DESYNC"
+	Auto        SyncMode = "AUTO"
+)
+
+func (h *handler) SetSyncMode(w http.ResponseWriter, r *http.Request) {
+	msg := "[handler] set sync mode"
+	ctx := context.Background()
+	vars := mux.Vars(r)
+
+	sessionID := r.Header.Get("Session")
+	username := vars["username"]
+	userID := user.GenerateUserID(username, sessionID)
+	syncMode := SyncMode(vars["syncMode"])
+
+	switch syncMode {
+	case ForceSync:
+		err := h.UserCollection.SetSynchronized(ctx, userID, true)
+		if err != nil {
+			handleError(w, http.StatusInternalServerError, log.ErrorLevel, msg, err, InternalServerError)
+			return
+		}
+		err = h.UserCollection.SetAutoSync(ctx, userID, false)
+		if err != nil {
+			handleError(w, http.StatusInternalServerError, log.ErrorLevel, msg, err, InternalServerError)
+			return
+		}
+		break
+	case ForceDesync:
+		err := h.UserCollection.SetSynchronized(ctx, userID, false)
+		if err != nil {
+			handleError(w, http.StatusInternalServerError, log.ErrorLevel, msg, err, InternalServerError)
+			return
+		}
+		err = h.UserCollection.SetAutoSync(ctx, userID, false)
+		if err != nil {
+			handleError(w, http.StatusInternalServerError, log.ErrorLevel, msg, err, InternalServerError)
+			return
+		}
+		break
```

Comment:
you can save some lines if you would assign the values to `synchronized` and `autosync` in the switch statement and use these variables in the db call